File: WePeiYang/app/src/main/java/com/twtstudio/retrox/wepeiyangrd/home/tools/ToolItemViewModel.java
package com.twtstudio.retrox.wepeiyangrd.home.tools;

import android.content.Context;
import android.content.Intent;
import android.databinding.ObservableField;
import android.databinding.ObservableInt;
import android.support.v7.app.AppCompatActivity;

import com.kelin.mvvmlight.base.ViewModel;
import com.kelin.mvvmlight.command.ReplyCommand;
import com.twtstudio.retrox.wepeiyangrd.base.BaseActivity;

/**
 * Created by retrox on 2017/1/15.
 */

public class ToolItemViewModel implements ViewModel {

    private Context mContext;

    private Class<? extends AppCompatActivity> targetAct;

    public final ObservableInt iconRes = new ObservableInt();

    public final ObservableField<String> title = new ObservableField<>();

    public final ReplyCommand clickCommand = new ReplyCommand(this::jump);

    public ToolItemViewModel(Context context, int iconres, String title, Class<? extends AppCompatActivity> activityClass) {
        mContext = context;
        this.iconRes.set(iconres);
        this.title.set(title);
        this.targetAct = activityClass;
    }

    private void jump(){
        Intent intent = new Intent(mContext,targetAct);
        mContext.startActivity(intent);
    }

}

File: WePeiYang/app/src/main/java/com/twt/service/home/tools/ToolItemViewModel.kt
package com.twt.service.home.tools

import android.content.Context
import android.content.Intent

import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.bumptech.glide.Glide
import com.twt.service.R

/**
 * Created by retrox on 2017/1/15.
 */


class ToolItem(val iconres: Int, val title: String, val targetAct: Class<out AppCompatActivity>)

class ToolItemViewHolder(private val mContext: Context, itemView: View) : RecyclerView.ViewHolder(itemView) {

    val layout: LinearLayout = itemView.findViewById(R.id.tool_linear_layout)
    val image: ImageView = itemView.findViewById(R.id.tool_image)
    val titleText: TextView = itemView.findViewById(R.id.tool_title)

    fun bind(toolItem: ToolItem) {
        toolItem.apply {
            Glide.with(mContext).load(iconres).into(image)
            titleText.text = title
            layout.setOnClickListener {
                val intent = Intent(mContext, targetAct)
                mContext.startActivity(intent)
            }
        }
    }

}
